schemaregistry/resource_schema.go

```diff
@@ -23,13 +24,36 @@ func resourceSchema() *schema.Resource {
 			StateContext: schema.ImportStatePassthroughContext,
 		},
 		CustomizeDiff: customdiff.ComputedIf("version", func(ctx context.Context, d *schema.ResourceDiff, meta interface{}) bool {
+
+			var schemaHasChange bool
 			oldState, newState := d.GetChange("schema")
-			newJSON, _ := structure.NormalizeJsonString(newState)
-			oldJSON, _ := structure.NormalizeJsonString(oldState)
-			schemaHasChange := newJSON != oldJSON
 
-			// explicitly set a version change on schema change and make dependencies aware of a
-			// version changed at `plan` time (computed field)
+			if schemaTypeStr, ok := d.Get("schema_type").(string); ok {
+				if strings.ToLower(schemaTypeStr) == "json" {
+					newJSON, _ := structure.NormalizeJsonString(newState)
+					oldJSON, _ := structure.NormalizeJsonString(oldState)
+					schemaHasChange = newJSON != oldJSON
+				} else if strings.ToLower(schemaTypeStr) == "avro" {
+					newJSON, _ := structure.NormalizeJsonString(newState)
+					oldJSON, _ := structure.NormalizeJsonString(oldState)
+					schemaHasChange = newJSON != oldJSON
+				} else if strings.ToLower(schemaTypeStr) == "protobuf" {
+					newProtoString, err := FormatProtobufString(newState.(string))
+					if err != nil {
+						// If theres an error diff should be true, indicating something is wrong?
```

Comment:
Not sure what to do if there is an error while formatting. Ideally i would propagate an error but it doesnt seem like the CustomizeDiff function accepts an error as a return.